src/client/dfs/obj.c

```diff
@@ -784,7 +807,18 @@ dfs_stat(dfs_t *dfs, dfs_obj_t *parent, const char *name, struct stat *stbuf)
 		oh = parent->oh;
 	}
 
-	return entry_stat(dfs, DAOS_TX_NONE, oh, name, len, NULL, true, stbuf, NULL);
+	if (dfs->dcache) {
+		dfs_obj_t *obj;
+
+		rc = dcache_find_insert_rel(dfs, parent, name, len, O_NOFOLLOW, &obj, NULL, stbuf);
+		if (rc)
+			return rc;
+		drec_decref(dfs->dcache, obj);
+	} else {
```

Comment:
Do we want to move this into entry_stat()?

Reply:
no it won't work. see ostat() where we assume entry_stat does not use cache.